Core/include/Acts/EventData/TrackStateProxy.hpp

```diff
@@ -656,9 +664,37 @@ class TrackStateProxy {
   ///       to another. Use the `projector` or `effectiveProjector` method if
   ///       you want to access the matrix.
   /// @return The projector bitset
-  ProjectorBitset projectorBitset() const {
+  /// @deprecated Use projector() instead
+  [[deprecated("use projector() instead")]] ProjectorBitset projectorBitset()
+      const {
+    return projectorSubspace().projectorBitset();
+  }
+
+  ProjectorMapping projectorMapping() const {
+    assert(has<hashString("projector")>());
+    return component<ProjectorMapping, hashString("projector")>();
+  }
+
+  SubspaceHelper<eBoundSize> projectorSubspace() const {
     assert(has<hashString("projector")>());
-    return component<ProjectorBitset, hashString("projector")>();
+    ProjectorMapping mapping = projectorMapping();
+    return SubspaceHelper<eBoundSize>(
+        {mapping.begin(), mapping.begin() + calibratedSize()});
+  }
+
+  void setProjector(const ProjectorMapping& proj) requires(!ReadOnly) {
+    component<ProjectorMapping, hashString("projector")>() = proj;
+  }
+
+  void setProjector(std::span<const BoundIndices> proj) requires(!ReadOnly) {
+    assert(has<hashString("projector")>());
+    assert(proj.size() == calibratedSize() && "Invalid projector size");
```

Comment:
I noticed that I just coupled `projector` and `calibrated` properties. Do you think we should generally do this @paulgessinger ? Or is there a case where a projector might exist but a measurement not?

Reply:
Good question. I guess the main concern would be that it's not obvious that you need to set them in a specific order or risk running into the assertion.